src/kv/store.h

```diff
@@ -199,25 +216,68 @@ namespace kv
     {
       std::lock_guard<SpinLock> mguard(maps_lock);
 
-      auto search = maps.find(name);
-      if (search != maps.end())
+      if (has_map_internal(name))
         throw std::logic_error("Map already exists");
-      auto replicated = true;
-      if (replicate_type == kv::ReplicateType::NONE)
+
+      auto result = std::make_shared<M>(
+        this, name, security_domain, is_map_replicated(name));
+      maps[name] = std::make_pair(NoVersion, result);
+      return *result;
+    }
+
+    std::shared_ptr<AbstractMap> get_map(
```

Comment:
Can we explicitly make this experimental with a comment (and keep it out of doxygen or mark it as such there too?).